Unable to start dev with Cloudflare adapter: `file already exists, mkdir './node_modules/.astro/'`

### Astro Info

```block
Astro                    v4.3.2
Node                     v21.6.1
System                   Linux (x64)
Package Manager          pnpm
Output                   server
Adapter                  @astrojs/cloudflare
Integrations             @astrojs/tailwind
                         @astrojs/solid-js
```


### Describe the Bug

Running `pnpm dev` stopped working, produces the following output (replaced absolute project path with '...')

```
10:31:45 [WARN] The currently selected adapter `@astrojs/cloudflare` is not compatible with the image service "Sharp".

 astro  v4.3.2 ready in 310 ms

┃ Local    http://localhost:4321/
┃ Network  use --host to expose

10:31:45 watching for file changes...
10:31:47 [ERROR] [UnhandledRejection] Astro detected an unhandled rejection. Here's the stack trace:
Error: EEXIST: file already exists, mkdir '/.../node_modules/.astro/'
    at mkdirSync (node:fs:1382:26)
    at LocalPagesRuntime.getCF (file:///.../node_modules/.pnpm/@astrojs+cloudflare@9.0.0_astro@4.3.2/node_modules/@astrojs/cloudflare/dist/utils/local-runtime.js:114:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async middleware (file:///.../node_modules/.pnpm/@astrojs+cloudflare@9.0.0_astro@4.3.2/node_modules/@astrojs/cloudflare/dist/index.js:67:36)
  Hint:
    Make sure your promises all have an `await` or a `.catch()` handler.
  Error reference:
    https://docs.astro.build/en/reference/errors/unhandled-rejection/
  Stack trace:
    at mkdirSync (node:fs:1382:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    [...] See full stack trace in the browser, or rerun with --verbose.
```

### What's the expected result?

`pnpm dev` works

### Workaround

Removing the directory fixes the problem permanently: `rm -r node_modules/.astro/`

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/eleven-books-push.md
+++ b/.changeset/eleven-books-push.md
@@ -2,4 +2,4 @@
 "@astrojs/cloudflare": patch
 ---
 
-Fixes an error which occured if the cache directory already exist, causing `astro dev` to fail
+Fixes an error where the `astro dev` command would fail if the cache directory already existed

--- a/.changeset/eleven-books-push.md
+++ b/.changeset/eleven-books-push.md
@@ -1,0 +1,5 @@
+---
+"@astrojs/cloudflare": patch
+---
+
+Fixes an error which occured if the cache directory already exist, causing `astro dev` to fail

--- a/packages/cloudflare/src/utils/local-runtime.ts
+++ b/packages/cloudflare/src/utils/local-runtime.ts
@@ -9,7 +9,7 @@ import type { AstroConfig, AstroIntegrationLogger } from 'astro';
 import type { Json, ReplaceWorkersTypes, WorkerOptions } from 'miniflare';
 import type { Options } from '../index.js';
 
-import { mkdirSync, readFileSync, statSync, writeFileSync } from 'fs';
+import { existsSync, mkdirSync, readFileSync, statSync, writeFileSync } from 'fs';
 import assert from 'node:assert';
 import { fileURLToPath } from 'url';
 import TOML from '@iarna/toml';
@@ -191,7 +191,9 @@ class LocalRuntime {
 		const CF_ENDPOINT = 'https://workers.cloudflare.com/cf.json';
 		if (!this.cfObject) {
 			this.cfObject = await fetch(CF_ENDPOINT).then((res) => res.json());
-			mkdirSync(this._astroConfig.cacheDir);
+			if (!existsSync(this._astroConfig.cacheDir)) {
+				mkdirSync(this._astroConfig.cacheDir);
+			}
 			writeFileSync(
 				fileURLToPath(new URL('cf.json', this._astroConfig.cacheDir)),
 				JSON.stringify(this.cfObject),

